tgui/packages/tgui/interfaces/MedicalRecords.js

```diff
@@ -314,89 +374,163 @@ const MedicalRecordsViewMedical = (_properties, context) => {
-    return <Box color="label">There are no Medbots.</Box>;
+    return <Box color="label">There are no Medibots.</Box>;
   }
-  return medbots.map((medbot, i) => (
-    <Collapsible key={i} open title={medbot.name}>
-      <Box px="0.5rem">
-        <LabeledList>
-          <LabeledList.Item label="Location">
-            {medbot.area || 'Unknown'} ({medbot.x}, {medbot.y})
-          </LabeledList.Item>
-          <LabeledList.Item label="Status">
-            {medbot.on ? (
-              <Fragment>
-                <Box color="good">Online</Box>
-                <Box mt="0.5rem">
+  return(
+    <Flex direction="column" height="100%">
+      <Section flexGrow="1" mt="0.5rem">
+        <Table className="MedicalRecords__list">
+          <Table.Row bold>
+            <Table.Cell>Name</Table.Cell>
+            <Table.Cell>Area</Table.Cell>
+            <Table.Cell>Status</Table.Cell>
+            <Table.Cell>Chemicals</Table.Cell>
+          </Table.Row>
+          {medbots
+            .map((medbot) => (
+              <Table.Row
+                key={medbot.id}
+                className={
+                  'MedicalRecords__listMedbot--' + medbot.on
+                }
+              >
+                <Table.Cell>
+                  <Icon name="medical" /> {medbot.name}
+                </Table.Cell>
+                <Table.Cell>{medbot.area || 'Unknown'} ({medbot.x}, {medbot.y})</Table.Cell>
+                <Table.Cell>{medbot.on
+                    ? (<Box color="good">Online</Box>)
+                    : (<Box color="average">Offline</Box>)}
+                </Table.Cell>
+                <Table.Cell>
                   {medbot.use_beaker
                     ? 'Reservoir: ' +
                       medbot.total_volume +
                       '/' +
                       medbot.maximum_volume
-                    : 'Using internal synthesizer.'}
-                </Box>
-              </Fragment>
-            ) : (
-              <Box color="average">Offline</Box>
-            )}
-          </LabeledList.Item>
-        </LabeledList>
-      </Box>
-    </Collapsible>
-  ));
+                    : 'Using internal synthesizer'}
+                </Table.Cell>
+            </Table.Row>
+          ))}
+        </Table>
+      </Section>
+    </Flex>
+  );
 };
 
 const MedicalRecordsNavigation = (_properties, context) => {
   const { act, data } = useBackend(context);
-  const { screen } = data;
+  const { screen, general } = data;
+  const [_sortId, setSortId] = useLocalState(context, 'sortId', 'name');
+  const [_sortOrder, setSortOrder] = useLocalState(context, 'sortOrder', true);
   return (
     <Tabs>
       <Tabs.Tab
         selected={screen === 2}
-        onClick={() => act('screen', { screen: 2 })}
-      >
-        <Icon name="list" />
+        onClick={
+          () => {
+            setSortId('name') // prevents trying to sort on null IDs
+            setSortOrder(true)
+            act('screen', { screen: 2 })
```

Comment:
Nesting logic for elements thats not used for navigation in the tabs seems like bad practice to me.

Reply:
This prevents the TGUI from bluescreening when switching between these two panels. The only other option I can think of is to have two different sortbutton classes for each screen, because they each edit the same variable (unless theres someway to pass a reference in to the sortbutton class). What would be preferable?

Reply:
You would need to make the sorbutton component take in some sort of uniqueid parameter which will set the `state`'s ID for localstate data storing. Otherwise you can just duplicate the code but change the state id name for each implementation.

Reply:
Yeah I've had no luck trying to get the sortbutton to use the variable thing... thing im gonna have to go with duplicates